lib/clients/BaseClient.py

```diff
@@ -70,6 +71,30 @@ def __init__(self, operation_name, configuration, directory_persistent, director
         self.__mounted_devices = []
         self.__devices = {}
 
+    def _get_credentials_from_credhub(self, configuration):
+        access_token = self.__getAccessToken(configuration)
+        params = {'name': configuration['credhub_key'],
+                  'current': 'true'}
+        headers = {'content-type': 'application/json',
+                   'authorization': 'bearer ' + access_token}
+        response = requests.get(url=configuration['credhub_url'].rstrip(
+            '/') + '/v1/data', headers=headers, params=params, verify=False)
+        credentials = response.json()
+        return credentials['data'][0]['value']
+
```

Comment:
Now since we are retrieving the latest credentials would it still return an array?

Reply:
Yes. Thats the reason, initially did not set that flag :-)